app/src/main/java/com/example/bikebuddy/RecordWorkoutDialog.java

```diff
@@ -0,0 +1,109 @@
+
+    //Buttons
+    Button buttonContinue;
+    Button buttonCancel;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_record_workout,container,false);
+
+        //Continue Button
+        //*************************************************************************************************************************************
+        buttonContinue = view.findViewById(R.id.buttonContinue);
+        buttonContinue.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ((FitnessFragment) getParentFragment()).continueWorkoutAction();
+                getDialog().dismiss();
+            }
+        });
+        //*************************************************************************************************************************************
+
+        //Cancel Button
+        //*************************************************************************************************************************************
+        buttonCancel = view.findViewById(R.id.buttonCancel);
+        buttonCancel.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                getDialog().dismiss();
+            }
+        });
+        //*************************************************************************************************************************************
+
+        //Bluetooth status
+        //*************************************************************************************************************************************
+        textViewBTStatus = view.findViewById(R.id.textDialogBTStatus);
+        //*************************************************************************************************************************************
+
+        //Location status
+        //*************************************************************************************************************************************
+        textViewLocationStatus = view.findViewById(R.id.textDialogLocationStatus);
+        //*************************************************************************************************************************************
+
+        updateSensorStatus();
+
+        return view;
+    }
+
+    private void updateSensorStatus()
+    {
+        if (MainActivity.isDeviceConnected)
+        {
+            textViewBTStatus.setText("Connected");
+        }
+        else
+        {
+            textViewBTStatus.setText("Not Connected");
+        }
+
+        if (isLocationEnabled(getContext()))
+        {
+            textViewLocationStatus.setText("Enabled");
+        }
+
+        else
+        {
+            textViewLocationStatus.setText("Not Enabled");
+        }
+    }
+
+    public static Boolean isLocationEnabled(Context context)
```

Comment:
There's a method called isMapsEnabled in Login Activity that checks whether the GPs is enabled on the device. You could've used that